### PR TITLE
initial genesis key op cert counters

### LINKED ISSUE #643: Genesis Key initial operational cert counters
When a stake pool registers itself, the operational certificate counter mapping is updated, adding `keyHash |-> 0`. During the overlay period, however, the genesis keys will be using the operational certs, but will not be in this mapping.

the `toShelley` function should not set the mapping as empty initially, but to a mapping of the genesis keys to zero.

### PR DESCRIPTION
**Pull request by JaredCorduan**
The genesis delegates (ie the cold keys that the genesis keys delegate to) are not registered like normal stake pool keys, and were not previously being given initial counters for the operational certificates. Now the `toShelley` method in the formal spec sets a counter of zero for each starting genesis key delegate.

Additionally, while making this change, I noticed a couple related things that also needed fixing:
* We should not allow two different genesis keys to delegate to the same cold key. There is now a check for this.
* When re-delegating a genesis key, the old delegate needs to be removed from the op cert counters.

closes #643

### CODE REVIEW

File: shelley/chain-and-ledger/executable-spec/src/STS/Ledgers.hs [hunk 1]
```diff
@@ -73,6 +73,7 @@ ledgersTransition = do
   let u''' = UTxOState utxo' dep fee (ppup, aup, favs', avs')
 
   let ds = _dstate dw''
+      ps = _pstate dw''
       fdms_    = _fdms ds
       Dms dms_ = _dms ds
       (fdms', curr) = Map.partitionWithKey (\(s, _) _ -> slot <= s) fdms_
```
Comment (mgudemann) [inline shelley/chain-and-ledger/executable-spec/src/STS/Ledgers.hs]: not part of the PR, but I think `curr` and `fdms'` is the wrong way, `fst` is the part that matches the predicate

File: shelley/chain-and-ledger/formal-spec/ledger.tex [hunk 2]
```diff
@@ -186,7 +188,14 @@ \section{Ledger State Transition}
       (\var{stkeys},~\var{rewards},~\var{delegations},~\var{ptrs},
       ~\var{fdms}\setminus\var{curr},~\var{dms}\unionoverrideRight\var{dms'})
       \\
-      \var{ls'}\leteq((\var{utxo},~\var{deposits},~\var{fees},~\var{us'}),~(\var{ds'},~\var{ps}))
+      \var{oldGenDelegs}\leteq\range((\dom\var{dms'})\restrictdom\var{dms})
+      \\
+      \var{cs'}\leteq(\var{cs}\setminus\var{oldGenDelegs})\unionoverrideRight
```
[the remark this answers is not in the document]
Comment (JaredCorduan) [reply shelley/chain-and-ledger/formal-spec/ledger.tex]: yes indeed! thanks

File: shelley/chain-and-ledger/executable-spec/src/STS/Ledgers.hs (hunk 1 above)
Comment (JaredCorduan) [reply shelley/chain-and-ledger/executable-spec/src/STS/Ledgers.hs]: it's true that `fst` is the part that matches the predicate, but I think it's still correct.

`s` is the lot assigned to a future delegation, and `slot` is the current slot. so if `slot <= s`, then this "future" delegation is legitimately in the future. otherwise, it's now a current (ie should be considered now) delegation.
